sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/FieldValueTypeSupplier.java

```diff
@@ -28,8 +29,10 @@
  */
 public interface FieldValueTypeSupplier extends Serializable {
   /**
-   * Return all the FieldValueTypeInformations. The returned list must be in the same order as
+   * Return all the FieldValueTypeInformations.
+   *
+   * <p>If the schema parameter is not null, then the returned list must be in the same order as
    * fields in the schema.
    */
-  List<FieldValueTypeInformation> get(Class<?> clazz, Schema schema);
+  List<FieldValueTypeInformation> get(Class<?> clazz, @Nullable Schema schema);
```

Comment:
There are few implementations where we override it to non-nullable that should be modified.

Looking into `FieldValueTypeSupplier#get` it feels that `@Nullable Schema schema` is only used to re-order fields. I would suggest moving this logic outside in usages where it's needed. Now it's confusing to see the case when `schema=null`, with moving sorting outside it wouldn't be needed.

What do you think?

Reply:
Since every single call site except for one needed this, moving the sorting to all call sites seems undesirable. Instead I split into two separate methods, one taking Schema one not, so that we don't need a @Nullable parameter

Reply:
Since given schema there is only a single correct way to perform sorting, do you mean to have:

```java
interface FieldValueTypeSupplier {
  default List< FieldValueTypeInformation> get(Class<?> clazz, Schema schema) { ... }
  List< FieldValueTypeInformation> get(Class<?> clazz);

}
```

It sounds good. Otherwise, we would have to duplicate sorting either in call-site or in each implementation of `FieldValueTypeSupplier`

Reply:
yes, that is exactly what I did, though the idea of a default method is a good one and .I'll switch to that.